### PR TITLE
Restructure

### LINKED ISSUE #324: Restructure buildpack
## Context
Per [PHP RFC 0001](https://github.com/paketo-buildpacks/rfcs/blob/main/text/php/0001-restructure.md) we are restructuring the PHP language family of buildpacks to make them more modular, up-to-date, and as a result, maintainable.

This buildpack currently provides `php`, the PHP binary distribution onto the `$PATH`, and was fairly recently [rewritten using the packit library](https://github.com/paketo-buildpacks/php-dist/pull/67).

## Issue
Since this buildpack is already using packit, and is already modular, this issue is primarily
 to do two things:

- [x] Per [the RFC](https://github.com/paketo-buildpacks/rfcs/blob/main/text/php/0001-restructure.md#buildpacks), this buildpack should generate a default configuration file (`php.ini`). Additionally, the `PHP_INI_SCAN_DIR` and `PHPRC` environment variables should be set. Check out [this conversation](https://github.com/paketo-buildpacks/php/issues/485#issuecomment-1018681581) for additional context.

- [x] Modify this buildpack to use [Buildpack API 0.7](https://github.com/buildpacks/spec/releases/tag/buildpack%2Fv0.7). It currently uses [Buildpack API 0.4](https://github.com/buildpacks/spec/releases/tag/buildpack%2Fv0.2), so some implementation updates will be needed. 

   Check out CNB API [migration guides](https://buildpacks.io/docs/reference/spec/migration/buildpack-api-0.4-0.5/) as well as buildpacks in other language families that perform similar tasks on API 0.7 (such as the [Node Engine buildpack](https://github.com/paketo-buildpacks/node-engine))

- [x] Add a `[[buildpack.licences]]` section to the `buildpack.toml`.

- [x] Support env var configuration instead of buildpack.yml (See [PHP buildpack RFC #2](https://github.com/paketo-buildpacks/rfcs/blob/main/text/php/0002-buildpack-yml-to-env-var.md))

## Acceptance
This issue can be considered accepted if:
1. The buildpack API is 0.7
2. I can add locally use this version of the PHP Dist buildpack in the PHP language family buildpack, and successfully build [all of the apps we currently support](https://github.com/paketo-buildpacks/php/tree/main/integration/testdata).

### PR DESCRIPTION
**Pull request by fg-j**
## Summary

Resolves #324 

## Use Cases

This buildpack now configures the default `php.ini`.
As of now, the `php.ini` files and associated environment variables configured in this buildpack will be overwritten by the PHP Web buildpack. This is expected behaviour that will go away when the PHP Web buildpack is broken down later in the rewrite work. 

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: build.go [hunk 1]
```diff
@@ -114,14 +163,60 @@ func Build(entries EntryResolver,
...
 
-		err = environment.Configure(phpLayer)
+		logger.Debug.Subprocess("Finding PHP extensions directory")
+		extensionsDir, err := files.FindExtensions(phpLayer.Path)
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+		logger.Debug.Break()
+
+		libDir := filepath.Join("lib")
```
Comment (thitch97) [inline build.go:173]: Is there a reason we do a `filepath.Join()` here instead of just `libDir := "lib"`?

Comment (fg-j) [reply build.go:173]: No reason. Good catch.

File: build.go [hunk 2]
```diff
@@ -114,14 +163,60 @@ func Build(entries EntryResolver,
...
+		logger.Subprocess("Generating default PHP configuration")
+		defaultConfig, buildpackConfig, err := files.WriteConfig(phpLayer.Path, context.CNBPath, PhpIniConfig{
+			IncludePath: strings.Join([]string{
+				filepath.Join(phpLayer.Path, "lib", "php"),
+				filepath.Join(context.WorkingDir, libDir),
+			}, string(os.PathListSeparator)),
+			ExtensionDir: extensionsDir,
+			// TODO: figure out where extensions and zendextensions arrays come from
+			// Do we even need to load extensions in the default INI file? Maybe better to simply require that folks add additional INI files?
+		})
```
[the remark this answers is not in the document]
Comment (thitch97) [reply build.go:189]: I don't have any context on why that might be the case, @dmikusa-pivotal might know more. As for your question in the comments, the suggestion of user-provided `.ini` files seems reasonable to me. Perhaps this is [already what users expect](https://github.com/paketo-buildpacks/php-web/issues/205#issuecomment-946100427).

**Review by thitch97 — DISMISSED** [review]

**Comment by thitch97** [comment]
I'm okay with merging these changes without changing the status quo regarding extensions and Zend extensions. Should any new contextual information or pain arise we can re-evaluate our approach.

**Pull request merged**